guava/src/com/google/common/math/IntMath.java

```diff
@@ -699,6 +699,29 @@ public static int mean(int x, int y) {
     // The alternative (x + y) >>> 1 fails for negative values.
     return (x & y) + ((x ^ y) >> 1);
   }
+  public static int digitsOfNFactorial(int n) {
+  //This method calculates digits number of number n's factorial without calculating 'n!'
+  //For example:for n=5--> 5!=120 --> returns 3
+		return (n == 0 || n == 1) ? 1 : (int) Math.floor(((n + 0.5)* Math.log(n) - n + 0.5 * Math.log(2 * Math.PI))	/ Math.log(10)) + 1;
+	}
+  public static int getNumbersOfDigit(int n) {
+  //This method calculates digits number of n
+  //For example:for n=4129--> returns 4
+		return (n == 0 || n == 1) ? 1 : (int) (Math.floor(Math.log10(n))) + 1;
+}
+public static boolean isPrime(int p) {
```

Comment:
If we're going to provide primality testing at all, we have a much more heavily optimized implementation Google-internally that we haven't open-sourced primarily because we weren't convinced anyone would use it.